core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1671,6 +1671,83 @@ public void enableDeduplicationWithUpsertKeys(LongList columnsIndexes) {
         }
     }
 
+    @Override
+    public void forceRemovePartitions(LongList partitions) {
+        long minTimestamp = txWriter.getMinTimestamp(); // partition min timestamp
+        long maxTimestamp = txWriter.getMaxTimestamp(); // partition max timestamp
+        boolean firstPartitionDropped = false;
+        boolean activePartitionDropped = false;
+
+        txWriter.beginPartitionSizeUpdate();
+        partitionRemoveCandidates.clear();
+
+        int removedCount = 0;
+        for (int i = 0; i < partitions.size(); i++) {
+            long timestamp = partitions.getQuick(i);
+            final int index = txWriter.getPartitionIndex(timestamp);
+            if (index < 0) {
+                LOG.debug().$("partition is already removed [path=").$substr(pathRootSize, path).$(", partitionTimestamp=").$ts(timestamp).I$();
+                continue;
+            }
+
+            removedCount++;
+            if (!activePartitionDropped) {
+                activePartitionDropped = timestamp == txWriter.getPartitionTimestampByTimestamp(maxTimestamp);
+                if (activePartitionDropped) {
+                    // Need to do this only once
+                    closeActivePartition(false);
+                }
+            }
+
+            firstPartitionDropped |= timestamp == txWriter.getPartitionTimestampByIndex(0);
+            columnVersionWriter.removePartition(timestamp);
+            txWriter.removeAttachedPartitions(timestamp);
+            partitionRemoveCandidates.add(timestamp);
```

Comment:
why are we filling up this list? Contents of the list is not used in this method and there is no comment to explain where is it used and why

Reply:
I added a comment:
```Java
            // Add the partition to partition remove list that can be deleted if there are no open readers
            // after the commit
            partitionRemoveCandidates.add(timestamp, txWriter.getPartitionNameTxn(index));
```